sdk/core/core/inc/az_result.h

```diff
@@ -34,6 +34,12 @@ enum {
     } \
   } while (0)
 
+#define AZ_BREAK_IF_FAILED(exp, result) \
```

Comment:
Could we refactor the code to use only `AZ_RETURN_IF_FAILED`. For clean-up resources I would propose to create additional functions. Something like this:

```c
AZ_NODISCARD az_result az_bar(...) {
   // some code
  AZ_RETURN_IF_FAILED(…);
  // some code
}

AZ_NODISCARD az_result az_foo(…) {
  // allocate/open something
  az_result const result = az_bar(…);
  // clean up something
  return result;
}
```

Reply:
I would rather rewrite to use if()s, than extract part of the highly related code to a separate function where it is not immediately visible/readable. Plus think about that function, the border of responsibility of part of code that it is doing is so artificial, that I'd have problem coming up with a function name.

Reply:
Done.